src/coreComponents/constitutive/solid/ElasticIsotropic.cpp

```diff
@@ -13,18 +13,18 @@
  */
 
 /**
- *  @file LinearElasticIsotropic.cpp
+ *  @file ElasticIsotropic.cpp
  */
 
-#include "LinearElasticIsotropic.hpp"
+#include "ElasticIsotropic.hpp"
 
 namespace geosx
 {
 using namespace dataRepository;
 namespace constitutive
 {
 
-LinearElasticIsotropic::LinearElasticIsotropic( std::string const & name, Group * const parent ):
+ElasticIsotropic::ElasticIsotropic( std::string const & name, Group * const parent ):
```

Comment:
An **elastic** behavior law is a behavior that walks the same path in any direction (which is not the case for plasticity where, above some threshold, the material does not come back using the same way.)

An **elastic** behavior can be non linear.

Therefore the current `LinearElasticIsotropic` is indeed a **linear elastic** model. Removing the `Linear` loses precision.

Reply:
This draft is actually just a clone of  #974, nothing is changed yet ^^

Reply:
@TotoGaz, yes, I am trying to get `white/feature/druckerPrager` in shape for a careful review, and will invite you.  On your point about `LinearElastic` it turns out this class implements three models: linear elastic, hypo-elastic, and hyper-elastic.  The latter two are not linear, so I removed the `Linear` moniker.  It also shortens names a bit.  It still makes sense to keep them all in one class, though, because they share the same member variables, data allocation routines, etc.  There are just subtle differences in the update functions.

Reply:
@sytuannguyen OK I see, but I did not notice that this was a draft (there is a specific `draft` mode for github's PRs). And you mentioned me as a reviewer, so 🤷 

OK @joshua-white Thx for the update.
Couldn't inheritance be used to share what needs to be while better separating the differences?

Reply:
@TotoGaz, yes, and I am definitely using this for some other models.  Let's revisit on the actual PR during the review.  By keeping them in one model we can keep our constitutive switchyard a bit shorter, but perhaps this is not the best justification.